### PR TITLE
feat: ROCm linux support

### PR DESCRIPTION
**Pull request by newfla**
The PR introduces the support for building the library with rocm/hipblas support.
The following changes have been made:
- Added  `hipblas` feature flag to `whisper-rs` and `whisper-rs-sys`
- ROCm install folder can be optionally specified by `env::HIP_PATH`. Defaults to `/opt/rocm`
- Gpu targets can be optionally specified by `env::AMDGPU_TARGETS`
- Panics when targeting windows

Unfortunately I wasn't able to build whisper.cpp on windows: generated vcxproj refuse to use hipcc and defaults to cl.exe 😣

### CODE REVIEW

**Review by tazz4843 — CHANGES_REQUESTED** [review]

File: Cargo.toml [hunk 1]
```diff
@@ -28,6 +28,7 @@ default = []
 raw-api = []
 coreml = ["whisper-rs-sys/coreml"]
 cuda = ["whisper-rs-sys/cuda", "_gpu"]
+hipblas = ["whisper-rs-sys/hipblas", "_gpu"]
```
[the remark this answers is not in the document]
Comment (newfla) [reply Cargo.toml:31]: The flag is strictly required:
When targeting hipblas, cmake defines both GGML_USE_HIPBLAS and GGML_USE_CUDA (see [CMakeLists.txt L402](https://github.com/ggerganov/whisper.cpp/blob/c7b6988678779901d02ceba1a8212d2c9908956e/CMakeLists.txt#L402)) 
In fact, the cuda and hip code are contained in the same src files, protected by precompiler directives (e.g. [mmq.cu L1076-L1120](https://github.com/ggerganov/whisper.cpp/blob/c7b6988678779901d02ceba1a8212d2c9908956e/ggml-cuda/mmq.cu#L1076-L1120))

File: README.md [hunk 2]
```diff
@@ -71,6 +71,7 @@ All disabled by default unless otherwise specified.
   **NOTE**: enabling this no longer guarantees semver compliance,
   as whisper-rs-sys may be upgraded to a breaking version in a patch release of whisper-rs.
 * `cuda`: enable CUDA support. Implicitly enables hidden GPU flag at runtime.
+* `hipblas`: enable ROCm/hipBLAS support. Only available on linux. Implicitly enables hidden GPU flag at runtime.
```
[the remark this answers is not in the document]
Comment (newfla) [reply README.md:74]: See previous comment

File: sys/build.rs [hunk 3]
```diff
@@ -58,6 +58,29 @@ fn main() {
...
     }
+    #[cfg(feature = "hipblas")]
+    {
+        println!("cargo:rustc-link-lib=hipblas");
+        println!("cargo:rustc-link-lib=rocblas");
+        println!("cargo:rustc-link-lib=amdhip64");
+        
+        cfg_if::cfg_if! {
+            if #[cfg(target_os = "windows")] {
+                panic!("Not supported yet!!!")
```
[the remark this answers is not in the document]
Comment (newfla) [reply sys/build.rs:69]: I've opened the issue https://github.com/ggerganov/whisper.cpp/issues/2202
Do you think it makes sense to include it in the error message?

File: Cargo.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (tazz4843) [reply Cargo.toml:31]: Was unaware of that, sounds good.

File: sys/build.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (tazz4843) [reply sys/build.rs:69]: It should at the least be noted that this is a ROCm issue and not act as if it's simply completely unsupported, adding a link to that upstream issue with a bit more detail in the panic message itself would be fine.

[the remark this answers is not in the document]
Comment (newfla) [reply sys/build.rs]: I've modified the panic message

**Review by tazz4843 — APPROVED** [review]

**Comment by tazz4843** [comment]
Thanks :)

**Pull request merged**